mongo/integration/unified/entity.go

```diff
@@ -401,6 +429,197 @@ func (em *EntityMap) addDatabaseEntity(entityOptions *entityOptions) error {
 	return nil
 }
 
+var (
+	envVarAwsAccessKeyID         = "AWS_ACCESS_KEY_ID"
+	envVarAwsSecretAccessKey     = "AWS_SECRET_ACCESS_KEY"
+	envVarAwsTempAccessKeyID     = "CSFLE_AWS_TEMP_ACCESS_KEY_ID"
+	envVarAwsTempSecretAccessKey = "CSFLE_AWS_TEMP_SECRET_ACCESS_KEY"
+	envVarAwsTempSessionToken    = "CSFLE_AWS_TEMP_SESSION_TOKEN"
+	envVarAzureTenantID          = "AZURE_TENANT_ID"
+	envVarAzureClientID          = "AZURE_CLIENT_ID"
+	envVarAzureClientSecret      = "AZURE_CLIENT_SECRET"
+	envVarGcpEmail               = "GCP_EMAIL"
+	envVarGcpPrivateKey          = "GCP_PRIVATE_KEY"
+)
+
+// getKmsCredential processes a value of an input KMS provider credential.
+// An empty document returns from the environment.
+// A string is returned as-is.
+func getKmsCredential(kmsDocument bson.Raw, credentialName string, envVar string, defaultValue string) (string, error) {
+	credentialVal, err := kmsDocument.LookupErr(credentialName)
+	if err == nil {
+		if str, ok := credentialVal.StringValueOK(); ok {
+			return str, nil
+		}
+
+		if doc, ok := credentialVal.DocumentOK(); ok {
```

Comment:
Optional: Similar to above, consider inverting this check to return early.

E.g.
```go
doc, ok := credentialVal.DocumentOK()
if !ok {
	return "", fmt.Errorf("expected String or Document for %v, got: %v", credentialName, credentialVal)
}
// ...
```

Reply:
Done.